packages/terra-dropdown-button/src/Button.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './Button.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The text that is displayed on the option.
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The function that is called when the option is clicked or enter/space is pressed with it focused.
+   */
+  onClick: PropTypes.func.isRequired,
+  /**
+   * @private
+   * Whether or not the option should show active styles
+   */
+  isActive: PropTypes.bool,
+  /**
+   * @private
+   * Whether or not the option should show keyboard focus styles
+   */
+  isFocused: PropTypes.bool,
+  /**
+   * @private
+   * Callback to tell the parent it should close the dropdown
+   */
+  requestClose: PropTypes.func,
+};
+
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+// Keyboard events are handled in _DropdownButton.jsx
+const Option = ({
+  label, onClick, isFocused, isActive, requestClose, ...customProps
+}) => (
+  <li>
+    <div
+      {...customProps}
+      onClick={() => { onClick(); requestClose(); }}
```

Comment:
This order of these functions might have the potential to cause issues if the onClick causes a focus change. Assuming these events fire synchronously if the action of the Button changes focus, what happens at the completion of requestClose? When the popup closes, does it attempt to return focus to the button?

Reply:
There's something strange about firing two functions in an anonymous block, but if the focus doesn't turn out to be an issue this should be fine.

Reply:
Although, the onClick is being called without any parameters. This might be worth talking about in scrum.

Reply:
A minor gotcha about passing the `event` parameter would be that sometimes the event is a `onClick` and sometimes a `onKeyDown`. The `event` parameter hadn't been excluded for any real reason, I just hadn't thought to include that.

Reply:
I'm second guessing the argument I made for calling this onClick, but I can't think of a name that feels more appropriate. I still think callback is pretty generic, but because this fires on key down too I'm a little unsure. How does Button handle this? Does it fire on click onKeyDown (enter / space)?

Reply:
terra-button allows the consumer to pass a `onClick` and/or `onKeyDown` listener. Being a HTML button (most of the time) `onClick` would get called when enter or space are pressed.

Another thing about the event is that true focus is actually on the list and not on the item itself, so `target` and possible others would have to be changed before the event is passed on.

Reply:
If we boil it down, this component is just a collection of actions that a user can select. All this component needs to do is render the actions and notify the consumer that an action was selected. The manner of that selection shouldn't factor into the API.

Can we change the prop from 'onClick' to 'onSelect' and universally execute the callback after the dropdown closes (due to keyup, clicks, etc.)? It'd be migrating away from a directly semantic `<button>` API, but given the complexities of this component, I feel like that is appropriate.

Reply:
> I still think callback is pretty generic, but because this fires on key down too I'm a little unsure. How does Button (terra-button) handle this? Does it fire onClick for onKeyDown (enter / space)?

terra-button calls onClick when enter or space is pressed, but that is just a feature of how native buttons work. The items inside the dropdown are only fancy `div`s as currently implemented.

Reply:
I am in favor of using `onSelect` to alleviate the confusion between the actions and abstract away the implementation details of how the action is invoked. I think we should avoid multiple callback functions if possible.

That being said, we should consider the implications of how this may affect future components with a similar behavior and navigate towards some level of consistency. The Menu component for example uses `onClick` to notify the user a selection has been made, regardless of if the selection was made from keyboard interaction.

https://engineering.cerner.com/terra-ui/#/components/terra-menu/menu/menu

Reply:
> The order of these functions might have the potential to cause issues if the onClick causes a focus change. Assuming these events fire synchronously if the action of the Button changes focus, what happens at the completion of requestClose? When the popup closes, does it attempt to return focus to the button?

I've confirmed that (regardless of order) if `onClick` tries to focus something that thing will get focused and then blurred as the dropdown unmounts. Even if I try to change the focus trap's props when the item gets clicked it unmounts before the props update. Fixing this would require changes in `focus-trap-react`.

Reply:
Focus fixed in b12e0ce243a3101ed65658c1c4314cf10002fa7d, renaming to `onSelect` soon